### DISCUSSION TITLE
Is nested ProtoBuf schema supported by eKuiper?

### DISCUSSION THREAD

**Discussion by thinkinds** [lf-edge/ekuiper#1584] [General]
I'm trying to use eKuiper to process ProtoBuf data. The official tutorial has been successfully performed, which has a simple one-layer schema: 
`message Book {  
  required string title = 1;   
  required int32 price = 2;  
}  
`
When I tried to using my nested schema like this:
`message VecRT{
	message SrvRT{	
		message MidRT{
			message rt{				
				optional int32 unit_id = 3;
				optional int32 chl_id = 4;
         ...
`
I got error `{"status": "running","source_mqtt_saas_test_0_records_in_total":0,"source_mqtt_saas_test_0_records_out_total":0,"source_mqtt_saas_test_0_process_latency_us":0,"source_mqtt_saas_test_0_buffer_length":0,"source_mqtt_saas_test_0_last_invocation":0,"source_mqtt_saas_test_0_exceptions_total":170,"source_mqtt_saas_test_0_last_exception":"Invalid data format, cannot decode \x1f\x8b\b\x00\x00\x00\x00\x00\x00\n\xac\x9dw|T\xc5\xdaǝt\xa2\x14\xb1\x80\x11\x05/\xc5\xd7\xeb\r\xa4\x1dR`o6!\x80\x05K,WQ\x91p\x95+(*\xa0H\xb3\xa4\xf7... with error decode failed: proto: bad wiretype","source_mqtt_saas_test_0_last_exception_time":"2022-12-28T03:50:16.034498","op_2_project_0_records_in_total":0,"op_2_project_0_records_out_total":0,"op_2_project_0_process_latency_us":0,"op_2_project_0_buffer_length":0,"op_2_project_0_last_invocation":0,"op_2_project_0_exceptions_total":0,"op_2_project_0_last_exception":"","op_2_project_0_last_exception_time":0,"sink_log_0_0_records_in_total":0,"sink_log_0_0_records_out_total":0,"sink_log_0_0_process_latency_us":0,"sink_log_0_0_buffer_length":0,"sink_log_0_0_last_invocation":0,"sink_log_0_0_exceptions_total":0,"sink_log_0_0_last_exception":"","sink_log_0_0_last_exception_time":0,"sink_mqtt_1_0_records_in_total":0,"sink_mqtt_1_0_records_out_total":0,"sink_mqtt_1_0_process_latency_us":0,"sink_mqtt_1_0_buffer_length":0,"sink_mqtt_1_0_last_invocation":0,"sink_mqtt_1_0_exceptions_total":0,"sink_mqtt_1_0_last_exception":"","sink_mqtt_1_0_last_exception_time":0}`
Does this mean that nested shema is not supported in eKuiper now?

**Comment by ngjaying** [comment]
Dear @thinkinds , eKuiper supportes nested message. Can you confirm that your data matches the schema? If possible, may you provide the pb and the data (hex) that produces this error? Thanks.
